consul/leader.go

```diff
@@ -562,11 +562,36 @@ func (s *Server) joinConsulServer(m serf.Member, parts *agent.Server) error {
 	}
 
 	// Attempt to add as a peer
-	addFuture := s.raft.AddPeer(raft.ServerAddress(addr))
-	if err := addFuture.Error(); err != nil {
-		s.logger.Printf("[ERR] consul: failed to add raft peer: %v", err)
+	minRaftProtocol, err := ServerMinRaftProtocol(s.serfLAN.Members())
+	if err != nil {
 		return err
 	}
+
+	if minRaftProtocol >= 2 && parts.RaftVersion >= 3 {
```

Comment:
Actually maybe this will be ok since you'll get an unsupported protocol version already from down below.